src/client/mod.rs

```diff
@@ -143,6 +143,58 @@ impl<T: Transport, L: LoginCredentials> TwitchIRCClient<T, L> {
             .await
     }
 
+    /// Ban a user with a optional reason in the given Twitch channel.
```

Comment:
Should include a documentation segment that says that no error is returned even if the target user does not exist, is already banned or the logged in user does not have permission to ban, or similar application-level errors